### PR TITLE
MQTT Discovery enhancements

### PR DESCRIPTION
**Pull request by pplucky**
- As [suggested here](https://www.home-assistant.io/docs/mqtt/discovery/), abbreviated all possible MQTT discovery configuration variable names in payload

- Added BSSID and Current Matrix Brightness to Matrix Info

- Added some Matrix Info data as Diagnostic Sensors (check picture below)

- Added Light entity (to allowing changing sleep mode and control brightness)

image

WebUI added info descriptions are not showing properly, please review/help:
image

### CODE REVIEW

File: src/PixelIt.ino [hunk 1]
```diff
@@ -1648,6 +1650,19 @@ String GetButtons()
...
+    JsonObject &root = jsonBuffer.createObject();
+
+    root["sleepMode"] = state;
+
+    String json;
+    root.printTo(json);
+
+    return json;
+}
+
```
[the remark this answers is not in the document]
Comment (pplucky) [reply src/PixelIt.ino]: Forgotten function, removed now.

File: docs/js/pixelit.js [hunk 2]
```diff
@@ -126,6 +126,7 @@ function RefershData(input) {
                     case "sleepMode":
                         value = (value ? "On" : "Off");
                         break;
+                    case "currentMatrixBrightness":
```
[the remark this answers is not in the document]
Comment (pplucky) [reply docs/js/pixelit.js]: Done

File: src/PixelIt.ino [hunk 3]
```diff
@@ -2222,34 +2248,39 @@ boolean MQTTreconnect()
...
-            "\"state_topic\":\"#MASTERTOPIC##STATETOPIC#\","
-            "\"unit_of_measurement\":\"#UNIT#\","
-            "\"value_template\":\"{{value_json.#VALUENAME#}}\","
-            "\"entity_category\":\"diagnostic\","
-            "\"icon\":\"mdi:#ICON#\","
+            "\"stat_t\":\"#MASTERTOPIC##STATETOPIC#\","
+            "\"unit_of_meas\":\"#UNIT#\","
+            "\"val_tpl\":\"{{value_json.#VALUENAME#}}\","
+            "\"ent_cat\":\"diagnostic\","
+            "\"ic\":\"mdi:#ICON#\","
```
[the remark this answers is not in the document]
Comment (pplucky) [reply src/PixelIt.ino:2344]: I suppose no breaking change, as using abbreviations is the same as using the longer strings, it is up to Home Assistant to understand what each abbreviation means.

**Comment by o0shojo0o** [comment]
Sorry, I'm currently sick with Corona, so it still takes a while with the merge. I ask for a little patience :)

**Comment by pplucky** [comment]
> Sorry, I'm currently sick with Corona, so it still takes a while with the merge. I ask for a little patience :)

No rush, get well soon.

**Comment by o0shojo0o** [comment]
Fits everything so  far, if you still resolved the conflicts I will gladly merge it :)  
image

**Comment by o0shojo0o** [comment]
@pplucky did you already have time to view the conflicts?
I would like to include your pull in the next version.

**Comment by pplucky** [comment]
> @pplucky did you already have time to view the conflicts?
> I would like to include your pull in the next version.

I did, but I could not fix it myself with my sparse git knowledge. I think I'll need to backup my changes, create a new branch from main and submit it again as a new PR...

Unless you have a better idea.

**Comment by pplucky** [comment]
> @pplucky I have tried to perform the merge, can you please validate once if I have taken over everything correctly?

Seems good for the changes I did, but my UI still doesn't look like yours above, don't know if I need to do something. My WebUI doesn't seem to get updated, I also cannot see the added functions...

**Comment by o0shojo0o** [comment]
If you want to see the dev version of the Ui you have to start it from the dev server -> https://github.com/o0shojo0o/PixelIt/wiki/Develop-on-WebUI

**Comment by pplucky** [comment]
> If you want to see the dev version of the Ui you have to start it from the dev server -> https://github.com/o0shojo0o/PixelIt/wiki/Develop-on-WebUI

In that case, I think we're good to go, from an MQTT discovery enhancement perspective.

**Pull request merged**